go-controller/pkg/util/node_annotations.go

```diff
@@ -415,8 +414,92 @@ func ParseNodeManagementPortAnnotation(node *kapi.Node) (int, int, error) {
+
+		cnode.Annotations, err = UpdateManagementPortMACAddressesAnnotation(cnode.Annotations, netName, macAddress)
+		if err != nil {
+			return fmt.Errorf("failed to update node %q management port mac address annotation %s for network %s",
+				node.Name, macAddress.String(), netName)
+		}
+		// It is possible to update the node annotations using status subresource
+		// because changes to metadata via status subresource are not restricted for nodes.
+		return kubeInterface.UpdateNodeStatus(cnode)
+	})
+	if resultErr != nil {
+		return fmt.Errorf("failed to update node %s annotation", node.Name)
+	}
+	return nil
+}
+
+// UpdateManagementPortMACAddressesAnnotation updates the OvnNodeManagementPortMacAddresses annotation for the network name 'netName'
+// with the provided MAC Address
+func UpdateManagementPortMACAddressesAnnotation(annotations map[string]string, netName string, macAddress net.HardwareAddr) (map[string]string, error) {
+	if annotations == nil {
+		annotations = map[string]string{}
+	}
+	err := updateNodeManagementPortMACAddressesAnnotation(annotations, netName, macAddress)
+	if err != nil {
+		return nil, err
+	}
+	return annotations, nil
+}
+
+// updateNodeManagementPortMACAddressesAnnotation updates the OvnNodeManagementPortMacAddresses annotation in
+// the 'annotations' map with the provided macAddress for the given netName.
+func updateNodeManagementPortMACAddressesAnnotation(annotations map[string]string, netName string, macAddress net.HardwareAddr) error {
+	var bytes []byte
+
+	// First get the all mac addresses for all existing networks
+	macAddressMap, err := parseNetworkMapAnnotation(annotations, OvnNodeManagementPortMacAddresses)
+	if err != nil {
+		if !IsAnnotationNotSetError(err) {
+			return fmt.Errorf("failed to parse node network management port annotation %q: %v",
+				annotations, err)
+		}
+		// in the case that the annotation does not exist
+		macAddressMap = map[string]string{}
+	}
+	macAddressMap[netName] = macAddress.String()
+
+	// if no networks left, just delete the network ids annotation from node annotations.
+	if len(macAddressMap) == 0 {
+		delete(annotations, OvnNodeManagementPortMacAddresses)
+		return nil
+	}
+
+	// Marshal all network ids back to annotations.
+	bytes, err = json.Marshal(macAddressMap)
+	if err != nil {
+		return err
+	}
+	annotations[OvnNodeManagementPortMacAddresses] = string(bytes)
+	return nil
+}
+
+// UpdateNodeManagementPortMACAddresses used only from unit tests
+func UpdateNodeManagementPortMACAddresses(node *kapi.Node, nodeAnnotator kube.Annotator, macAddress net.HardwareAddr, netName string) error {
+	macAddressMap, err := parseNetworkMapAnnotation(node.Annotations, OvnNodeManagementPortMacAddresses)
+	if err != nil {
+		if !IsAnnotationNotSetError(err) {
+			return fmt.Errorf("failed to parse node network management port annotation %q: %v",
+				node.Annotations, err)
+		}
+		// in the case that the annotation does not exist
+		macAddressMap = map[string]string{}
```

Comment:
maybe is better to handle this the opposite way so we don't really need the comment 

```golang
                if IsAnnotationNotSetError(err) {
                        macAddressMap = map[string]string{}
                } else {
			return fmt.Errorf("failed to parse node network management port annotation %q: %v",
				node.Annotations, err)
		}
```

Reply:
hmm I actually just copied these bits that are done the same way in `updateNetworkIDsAnnotation` which I think should be fine..

Reply:
So not a must, if you decide not to do this then I am fine too.

Reply:
ack didn't change this now.. keeping it as is.